File: api/src/PersonalFinance.Domain/Entities/Transaction.cs
namespace PersonalFinance.Domain.Entities
{
    public class Transaction
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; } = string.Empty;
        public string Remarks { get; set; } = string.Empty;
        public string Flow { get; set; } = "DB"; // "DB" = Debit, "CR" = Credit
        public string Type { get; set; } = "Expense"; // "Expense" or "Income"
        public string Category { get; set; } = "Untracked Expense";
        public string Wallet { get; set; } = string.Empty; // e.g., "BCA", "SeaBank"
        public decimal AmountIdr { get; set; }
        public string Currency { get; set; } = "IDR";
        public decimal? ExchangeRate { get; set; } // For foreign currencies
    }
}

File: apps/frontend/src/types/Transaction.ts

export interface Transaction {
  id: string;
  date: string;
  description: string;
  amount: number;
  flow?: string;
  type: 'income' | 'expense' | 'transfer';
  category: string;
  bank: string;       // wallet text from AI service (transient)
  accountId?: string; // uuid of the linked account
  balance?: number;
  statementBalance?: number | null; // tie-breaker for dedup — must survive preview→submit round-trip
  isRecurring?: boolean;
  isDuplicate?: boolean;
}

export interface CategoryRule {
  id: number | string;
  keyword: string;
  category: string;
  type: 'income' | 'expense' | 'transfer';
  keywordLength?: number;
}

export interface MonthlySum {
  month: string;
  income: number;
  expenses: number;
  balance: number;
}

export type BankId = 'BCA' | 'Superbank' | 'NeoBank' | 'Wise' | 'Jago';

export interface WalletBalance {
  bankId: BankId;
  label: string;
  balance: number;
  currency: 'IDR' | 'USD';
  delta30d: number;
  sparkline: number[];
}
